### PR TITLE
remove ufraw and the gimp plugin

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/97946

```
commit 4e4c61d46e25c44cd6fc6ba2fbf5676dd2e03705
Author: Florian Klink <flokli@flokli.de>
Date:   Mon Jan 11 02:04:20 2021 +0100

    ufraw: remove
    
    This is broken and unmaintained. nufraw seems to be a drop-in
    alternative.
    
    Closes https://github.com/NixOS/nixpkgs/issues/97946.

commit 07e6b3bdb3ae32ec9a82c728cb25dffd75cf1736
Author: Florian Klink <flokli@flokli.de>
Date:   Mon Jan 11 02:03:35 2021 +0100

    gimpPlugins.nufraw: expose, remove broken gimpPlugins.ufraw

commit f09dc88cc727a83016c81f243179ff06e7878785
Author: Florian Klink <flokli@flokli.de>
Date:   Mon Jan 11 02:02:55 2021 +0100

    nufraw: provide gimp plugin
    
    This introduces a new "gimpPlugin" output, similar to how it was done in
    ufraw.

```

I didn't run the nufraw plugin, but it builds fine.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/nufraw/default.nix [hunk 1]
```diff
@@ -18,8 +19,11 @@
 , zlib
 
 , addThumbnailer ? false
+, withGimpPlugin ? true
 }:
 
+assert withGimpPlugin -> gimp != null;
+
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/nufraw/default.nix:26]:
```suggestion
```

**Comment by austinbutler** [comment]
Tried this out, it fails to load several different RAW files (Canon, Nikon, Panasonic).

```
/nix/store/3mr77qcnyarp9439xz87jfx0v691f9i5-gimp-with-plugins-2.10.22/lib/gimp/2.0/plug-ins/nufraw-gimp: fatal error: Segmentation fault
```

Loading directly, outside of GIMP, also fails.

```
[nix-shell:~/Documents/]$ nufraw ~/Pictures/P1000252.RW2
Segmentation fault (core dumped)
```

GIMP will default to using `darktable` if that's installed (I had to comment it out temporarily in my `configuration.nix` to test `nufraw`), and `darktable` works flawlessly, so I wonder if it's even worth exposing `nufraw` as a GIMP plugin. I think RawTherapee will also be used by GIMP for RAW.

**Comment by asbachb** [comment]
I guess you're right. The main functionality seems to fail. On only tested with:

`nufraw-batch --size 10 --out-type=png --noexif --output=preview.png --embedded-image P1250767.RW2`

That works unless you're removing that `--embedded-image` flag. I guess when it's processing the raw footage the segmentation fault occurs.

**Comment by flokli** [comment]
So nufraw is currently entirely broken too?

In that case, I could trim that PR down, not include any nufraw changes and direct users to darktable?

**Comment by asbachb** [comment]
@flokli Yes I think you should not link anything against `nufraw` for now.

**Comment by asbachb** [comment]
I could get rid of the segmentation fault by applying the archlinux patch: https://github.com/asbachb/nixpkgs/tree/bugfix/nufraw

`nufraw-batch` now does not crash with an segfault but it creates just black images.

**Comment by asbachb** [comment]
So `lensfun` seems somehow broken or incompatible. When adding flag `--lensfun=none` the preview is generated as expected.

**Comment by flokli** [comment]
I pushed a new version, now recommending Darktable, and not adding the nufraw changes.

The commit message still links to this PR, so someone interested in getting nufraw to work can pick it up.

**Comment by austinbutler** [comment]
Result of `nixpkgs-review pr 108993` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by austinbutler — APPROVED** [review]

**Pull request merged**